### DISCUSSION TITLE
Downloading more than 2 streams

### DISCUSSION THREAD

**Discussion by wewerak47** [Tyrrrz/YoutubeExplode#692] [Q&A] [answered]
Why we cannot download more than 2 streams out of the box when the rest of the code allows it.
https://github.com/Tyrrrz/YoutubeExplode/blob/7f21ffea6b7f235a5bd35f15106313e7b1fd204c/YoutubeExplode.Converter/Converter.cs#L180-L181
https://github.com/Tyrrrz/YoutubeExplode/blob/7f21ffea6b7f235a5bd35f15106313e7b1fd204c/YoutubeExplode.Converter/Converter.cs#L40-L41
I tried cloning the repo, removing the if statement and the downloaded video has all the video and audio streams.
It will be great if the video streams have quality as a name.
![image](https://user-images.githubusercontent.com/33736792/222931846-09220cc4-b775-43d7-adfd-d9b980010b4b.png)

**Comment by Tyrrrz** [comment]
I haven't considered that use case. Does the player let you switch between different combination (audio/video) of streams at will? Not sure I ever encountered a video file like that.

**Reply by wewerak47** [reply]
I am trying that in VLC and i can switch between them. Haven't tried different player. 
The usecase for me is that We have a lot of different hardware we want to play videos on. We want to play 4k on PC, FullHD on raspberry pi and so on. But we are not limited by file size.

**Comment by wewerak47** [comment]
I have made some changes. You can review them in my fork.
https://github.com/wewerak47/YoutubeExplode/commit/af43ad30c3a36d6991ad93db5b99fdd0581a948f

**Comment by Tyrrrz** [comment, marked as answer]
Implemented it in: https://github.com/Tyrrrz/YoutubeExplode/commit/cbc54cf3184a5bacef389515da3b154d896397dd
Will release later this week
